test/shience.test/ScienceTests.cs

```diff
@@ -0,0 +1,142 @@
+            }
+            
+            [Fact]
+            public void DefaultComparerReturnsTrueWithSameResultOnPrimitives()
+            {
+                var science = Shience.New<bool>("DefaultComparerReturnsTrueWithSameResultOnPrimitives");
+
+                var result = science.Test(control: () => true, candidate: () => true).Execute();
+
+                Assert.Equal(true, PublishingResults.TestNamesWithResults["DefaultComparerReturnsTrueWithSameResultOnPrimitives"]);
+            }
+
+            [Fact]
+            public void DefaultComparerReturnsFalseWithDifferentResultOnPrimitives()
+            {
+                var science = Shience.New<bool>("DefaultComparerReturnsFalseWithDifferentResultOnPrimitives");
+
+                var result = science.Test(control: () => true, candidate: () => false).Execute();
+
+                Assert.Equal(false, PublishingResults.TestNamesWithResults["DefaultComparerReturnsFalseWithDifferentResultOnPrimitives"]);
+            }
+
+            [Fact]
+            public void DefaultComparerReturnsTrueWithSameResultOnObject()
+            {
+                var science = Shience.New<TestNumber>("DefaultComparerReturnsTrueWithSameResultOnObject");
+
+                var result = science.Test(
+                    control: () => new TestNumber(1),
+                    candidate: () => new TestNumber(1)).Execute();
+
+                Assert.Equal(true, PublishingResults.TestNamesWithResults["DefaultComparerReturnsTrueWithSameResultOnObject"]);
+            }
+
+            [Fact]
+            public void DefaultComparerReturnsFalseWithDifferentResultOnObject()
+            {
+                var science = Shience.New<TestNumber>("DefaultComparerReturnsFalseWithDifferentResultOnObject");
+
+                var result = science.Test(
+                    control: () => new TestNumber(1),
+                    candidate: () => new TestNumber(2)).Execute();
+
+                Assert.Equal(false, PublishingResults.TestNamesWithResults["DefaultComparerReturnsFalseWithDifferentResultOnObject"]);
+            }
+
+            [Fact]
+            public void ComparerFuncReturnsCorrectTrueResult()
+            {
+                var science = Shience.New<bool>("ComparerFuncReturnsCorrectTrueResult");
+
+                var result = science.Test(control: () => true,
+                    candidate: () => true)
+                    .WithComparer((a, b) => a == b)
+                    .Execute();
+
+                Assert.Equal(true, PublishingResults.TestNamesWithResults["ComparerFuncReturnsCorrectTrueResult"]);
+            }
+
+            [Fact]
+            public void ComparerFuncReturnsCorrectFalseResult()
+            {
+                var science = Shience.New<bool>("ComparerFuncReturnsCorrectFalseResult");
+
+                var result = science.Test(control: () => true,
+                    candidate: () => false,
+                    comparer: (a, b) => a == b).Execute();
```

Comment:
There's no overload of `Test` with a comparer now. This needs to use `WithComparer`.